support-lib/jni/djinni_support.cpp

```diff
@@ -371,6 +371,15 @@ jstring jniStringFromUTF8(JNIEnv * env, const std::string & str) {
     return res;
 }
 
+jstring jniStringFromWString(JNIEnv * env, const std::wstring & str) {
+
+    std::u16string utf16(str.begin(), str.end());;
```

Comment:
+1. Ideally we'd have a compile-time check to see whether `wchar_t` is of size 2 or 4 and behave appropriately either way. Failing that, there should at least be a `static_assert` with a clear message (e.g. "UCS-4 builds are currently unsupported")